portfolio/src/main/java/com/google/sps/servlets/ImageFormServlet.java

```diff
@@ -0,0 +1,68 @@
+package com.google.sps.servlets;
+
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.util.UUID;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.MultipartConfig;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+
+@WebServlet("/upload")
+@MultipartConfig()
+public class ImageFormServlet extends HttpServlet {
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) 
+        throws ServletException, IOException {
+        //Generate unique ID
+        UUID uuid = UUID.randomUUID();
+        //Get message & files, potentially useful when storing images with tags?
+        String message = request.getParameter("message");
+
+        //Get the actual file
```

Comment:
Comments like this are sometimes helpful but if the code is self-explanatory (because you wrote it cleanly here), they're typically not needed.